If an entered field is an int, validate it and return a user-level error

jefft opened the pull request:
Fixes #1265

The validation happens way down in the DB_Object class. That is also where `allow_empty` and `max_length` constraints are checked, so why not basic type checking too.

Perhaps other types besides 'int' should be validated in the same place, like 'datetime'. However I don't know how to trigger saving of a datetime. Without knowing how to trigger that codepath, I wouldn't want to risk breaking stuff. So validation is for int only.

radarhere commented on include/db_object.class.php:
```diff
@@ -612,6 +612,17 @@ public function validateFields()
 				trigger_error('The value for '.array_get($details, 'label', $id).' is not the correct length (must be exactly '.$details['fixed_length'].' characters)', E_USER_NOTICE);
 				$res = FALSE;
 			}
+			if ($res) { // If not already errored out..
```
```suggestion
			if ($res) { // If not already errored out...
```

jefft commented:
Actually, this breaks import of previously allowed CSV:

```
Title,Length_Mins,CCLI_Number
My Song2,0,,
```

CCLI_Number is not required, but now the line breaks with 'CCLI Number value "" is not a valid integer'

I'll work on a fix.

jefft commented:
I noticed that, without this patch, Jethro breaks under PHP 7.4 but correctly catches the CSV error when running PHP 8.4:

<img width="731" height="416" alt="image" src="https://github.com/user-attachments/assets/75f9d2f3-551e-47dd-adfc-1f97d94f5155" />

The int-checking code is:

https://github.com/tbar0970/jethro-pmm/blob/a41fca1131f05d7ec8d3e24b7821a1caf8c5aae9/include/db_object.class.php#L549-L560

This convoluted "is this string an int" code doesn't work properly, because PHP is an insane language where `0 == ':'` evaluates to `true`:

<img width="282" height="188" alt="image" src="https://github.com/user-attachments/assets/b4fdd423-1c81-4c69-9091-8b59e43f78e2" />

at least until PHP 8 implemented [PHP RFC: Saner string to number comparisons](https://wiki.php.net/rfc/string_to_number_comparison), and changed the evaluation to false:

<img width="204" height="191" alt="image" src="https://github.com/user-attachments/assets/8d70d20c-6dce-4f32-99f9-03ef0dbbc42a" />

I have changed my patch to use `filter_var($strval, FILTER_VALIDATE_INT)`, which seems to work in PHP 7.4 and 8.x:

<img width="740" height="171" alt="image" src="https://github.com/user-attachments/assets/6920a3dd-5683-45f2-994e-0fab0439b59c" />

jefft commented:
I've just force-pushed a fix - "0" was being classed as invalid.
